src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -95,6 +95,106 @@ public void Kill()
             }
         }
 
+        /// <summary>
+        /// Makes a best-effort attempt to kill all descendant (child, grandchild, etc.) processes.
+        /// </summary>
+        private void KillTree()
+        {
+            try
+            {
+                /*
+                   Causes this object instance to hold a handle to the process. While held, the process's PID won't be reused 
+                   and its children can be enumerated. These behaviors hold true even if the process is subsequently terminated.
+                */
+                OpenProcessHandle();
+
+                // Kill the process, so that no further children can be created.
+                Kill();
+            }
+            catch
+            {
+                // Made a best-effort attempt which failed (perhaps because the process is already dead), so give up.
+                return;
+            }
+
+            KillChildren(GetChildProcesses());
+        }
+
+        /// <summary>
+        /// Returns all immediate child processes.
+        /// </summary>
+        private IReadOnlyList<Process> GetChildProcesses()
+        {
+            var childProcesses = new List<Process>();
+
+            foreach (Process possibleChildProcess in GetProcesses())
+            {
+                var keep = false;
+
+                try
+                {
+                    try
+                    {
+                        /*
+                           Force the process object to hold a handle to the process. Ensures that if the process dies while we're working
+                           with it, it won't be reused. This way, any process we pass back is guaranteed to be an actual child, not a 
+                           reference to a new process that happens to have the same id as a deceased child.
+                        */
+                        possibleChildProcess.OpenProcessHandle();
```

Comment:
No, I don't believe so, because `Process` holds a reference to the handle and ensures that it is disposed when the process is disposed (Process's `Dispose()` calls [`Close()`](https://github.com/dotnet/corefx/blob/8b66ba0fe5555e4c69d502cf69b6640f81da6001/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L846) which disposes the handles). 

`GetChildProcess()` makes sure to dispose all non-children; `KillChildren()` disposes all children--so between the two all of the child handles opened here are disposed.